src/linear_algebra/determinant.lean

```diff
@@ -404,31 +402,19 @@ begin
     { exact h2.left x },
     { exact h2.right x }},
   { intros σ hσ,
-    have h1 : ∀ (x : m ⊕ n), (∃ (a : m), sum.inl a = x) → (∃ (a : m), sum.inl a = σ x),
-    { rintros x ⟨a, ha⟩,
-      rw ← ha,
-      exact (@mem_preserving_A σ).mp hσ a },
-    have h2 : ∀ (x : m ⊕ n), (∃ (b : n), sum.inr b = x) → (∃ (b : n), sum.inr b = σ x),
-    { rintros x ⟨b, hb⟩,
-      rw ← hb,
-      exact (perm_on_inl_iff_perm_on_inr σ).mp ((@mem_preserving_A σ).mp hσ) b },
-    let σ₁' := subtype_perm_of_fintype σ h1,
-    let σ₂' := subtype_perm_of_fintype σ h2,
-    let σ₁ := perm_congr (equiv.set.range (@sum.inl m n) sum.injective_inl).symm σ₁',
-    let σ₂ := perm_congr (equiv.set.range (@sum.inr m n) sum.injective_inr).symm σ₂',
-    use [⟨σ₁, σ₂⟩, finset.mem_univ _],
-    ext,
-    cases x with a b,
-    { rw [equiv.sum_congr_apply, sum.map_inl, perm_congr_apply, equiv.symm_symm,
-        set.apply_range_symm (@sum.inl m n)],
-      erw subtype_perm_apply,
-      rw [set.range_apply, subtype.coe_mk, subtype.coe_mk] },
-    { rw [equiv.sum_congr_apply, sum.map_inr, perm_congr_apply, equiv.symm_symm,
-        set.apply_range_symm (@sum.inr m n)],
-      erw subtype_perm_apply,
-      rw [set.range_apply, subtype.coe_mk, subtype.coe_mk] }},
-  { intros σ h0 hσ,
-    obtain ⟨a, ha⟩ := not_forall.mp ((not_congr (@mem_preserving_A σ)).mp hσ),
+    erw [set.mem_to_finset, monoid_hom.mem_range] at hσ,
+    obtain ⟨σ₁₂, hσ₁₂⟩ := hσ,
+    use σ₁₂,
+    rw ←hσ₁₂,
+    simp },
+  { intros σ hσ hσn,
+    have h1 : ¬ (∀ x, ∃ y, sum.inl y = σ (sum.inl x)),
+    { by_contradiction,
+      rw set.mem_to_finset at hσn,
+      apply absurd (mem_sum_congr_hom_range_of_perm_maps_to_inl _) hσn,
+      rintros x ⟨a, ha⟩,
+      rw [←ha], exact h a },
+    obtain ⟨a, ha⟩ := not_forall.mp h1,
     generalize hx : σ (sum.inl a) = x,
     cases x with a2 b,
```

Comment:
Does it work if you combine these lines to
```suggestion
    cases hx : σ (sum.inl a) with a2 b,
```
or am I forgetting how these things work?

Reply:
You are right. It works.